### PR TITLE
Unify AstRelationName/AstTypeName; introduce class AstQualifiedName

### PR DESCRIPTION
**Pull request by b-scholz**
Currently, we have two classes AstRelationName and AstTypeName defining qualified names for relations and types in our component model. 
This pull-request unifies both classes and names the new class AstQualfiedName. 

This fixes partly issue #1255.

### CODE REVIEW

File: src/AstQualifiedName.h [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ * Qualified names are used for types and relations.
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "Util.h"
+
+#include <algorithm>
+#include <string>
```
Comment (mmcgr) [inline src/AstQualifiedName.h]: Add sstream include

File: src/AstQualifiedName.h [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+#include <utility>
```
Comment (mmcgr) [inline src/AstQualifiedName.h]: Is utility used? I'm not seeing where.

File: src/AstQualifiedName.h [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    bool operator==(const AstQualifiedName& other) const {
+        return qualifiers == other.qualifiers;
+    }
+
+    bool operator!=(const AstQualifiedName& other) const {
+        return !(*this == other);
+    }
+
+    bool operator<(const AstQualifiedName& other) const {
+        return std::lexicographical_compare(
```
Comment (mmcgr) [inline src/AstQualifiedName.h]: I think this is equivalent to `return qualifiers < other.qualifiers;`, which would be simpler and not require the algorithm include

Comment (b-scholz) [reply src/AstQualifiedName.h]: No - unfortunately, this gives a different semantics and we fail to execute benchmarks such as name_clash.dl.

**Pull request merged**